tests/Unit/Options/Python/Test_ExtractInputSourceYAMLFromH5.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre.Options.ExtractInputSourceYAMLFromH5 import *
+
+import spectre.Informer as spectre_informer
+import unittest
+import os
+
+# For Py2 compatibility
+try:
+    unittest.TestCase.assertRaisesRegex
+except AttributeError:
+    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
+
+
+class TestExtractInputSourceYAMLFromH5(unittest.TestCase):
+    def test_read_input_source_from_h5(self):
+        expected_input_source = """# Distributed under the MIT License.
```

Comment:
can't you read the yaml file to a string in python instead of copying the contents which will undoubtedly change in the future?